crates/ruma-events/src/call/member.rs

```diff
@@ -0,0 +1,386 @@
+//! Types for matrix rtc state events ([MSC3401]).
+//!
+//! This implements a newer/updated version of MSC3401
+//! [MSC3401](https://github.com/matrix-org/matrix-spec-proposals/pull/3401)
+
+use std::time::Duration;
+
+use as_variant::as_variant;
+use ruma_common::{MilliSecondsSinceUnixEpoch as Timestamp, OwnedUserId};
+use ruma_macros::EventContent;
+use serde::{Deserialize, Serialize};
+use tracing::warn;
+
+/// The member state event for a matrixRTC session
+///
+/// This is the object containing all the data related to a matrix users participation in a
+/// matrixRTC session. It consists of memberships/sessions.
+#[derive(Clone, Debug, Serialize, Deserialize, EventContent, PartialEq)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
```

Comment:
Please use this `cfg_attr` for other types too, instead of just marking them a `non_exhaustive`.